plugins/org.komodo.core/src/org/komodo/repository/RepositoryImpl.java

```diff
@@ -499,214 +492,66 @@ private KomodoObject create( final UnitOfWork uow,
-                hasKeywords = true;
-
-                if (hasArtifactTypes) {
-                    whereClause.append(" AND "); //$NON-NLS-1$
-                }
-
-                // "CONTAINS(ARTIFACT.[tko:description],'foo' 'bar')"
-                whereClause.append("CONTAINS(").append(ALIAS).append(".[").append(LibraryComponent.DESCRIPTION).append("],"); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-
-                if (searchCriteria == null) {
-                    searchCriteria = KeywordCriteria.getDefault();
-                }
-
-                boolean firstTime = true;
-
-                for (final String keyword : keywords) {
-                    if (!firstTime && (KeywordCriteria.ANY == searchCriteria)) {
-                        whereClause.append(" OR "); //$NON-NLS-1$
-                    }
-
-                    if (KeywordCriteria.NONE == searchCriteria) {
-                        whereClause.append('-');
-                    }
-
-                    whereClause.append('\'').append(keyword).append('\'');
-                    firstTime = false;
-                }
-            }
-
-            // construct query
-            final String COLUMNS = JcrLexicon.PRIMARY_TYPE.getString() + ", " + LibraryComponent.DESCRIPTION; // TODO need other columns //$NON-NLS-1$
-            String selectStmt = "SELECT " + COLUMNS + " FROM [" + LibraryComponent.MIXIN_TYPE + "] " + ALIAS + ' '; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-
-            if (hasArtifactTypes || hasKeywords) {
-                selectStmt += whereClause;
-            }
-
-            LOGGER.debug("Artifact query: ", selectStmt); //$NON-NLS-1$
-
-            // run query
-            final Session session = getSession(transaction);
-            final QueryManager queryMgr = session.getWorkspace().getQueryManager();
-            final Query query = queryMgr.createQuery(selectStmt, Query.JCR_SQL2);
-            final QueryResult resultSet = query.execute();
-
-            // create results
-            ArtifactDescriptor[] result = null;
-            final NodeIterator itr = resultSet.getNodes();
-
-            if (itr.hasNext()) {
-                result = new ArtifactDescriptor[(int)itr.getSize()];
-                int i = 0;
-
-                while (itr.hasNext()) {
-                    final Node node = itr.nextNode();
-
-                    // description
-                    String description = null;
-
-                    if (node.hasProperty(LibraryComponent.DESCRIPTION)) {
-                        description = node.getProperty(LibraryComponent.DESCRIPTION).getString();
-                    } else {
-                        description = Messages.getString(Messages.Komodo.NO_ARTIFACT_DESCRIPTION, node.getName());
-                    }
-
-                    // version
-                    String version = null;
+        ObjectSearcher searcher = new ObjectSearcher(this);
+        searcher.addFromType(type, "k1"); //$NON-NLS-1$
```

Comment:
k1 is simply an alias as in the syntax SELECT * FROM [nt:unstructured] k1 where k1.[name] = "Bob". Aliases are required if a where clause is to be used. I can document it but is always noted in the javadoc of the method.